sympy/physics/mechanics/kane.py

```diff
@@ -329,9 +329,9 @@ def _form_frstar(self, bl):
         # components of rigid bodies. The inner most list is the list of
         # partial velocities.
         def get_partial_velocity(body):
-            if isinstance(body, RigidBody):
+            if isinstance(body, RigidBody) and body.is_rigidbody:
                 vlist = [body.masscenter.vel(N), body.frame.ang_vel_in(N)]
-            elif isinstance(body, Particle):
+            elif isinstance(body, Particle) and body.is_particle:
```

Comment:
Why is KanesMethod modified here? KanesMethod only accepts RigidBody objects, not Body objects. You should preprocess in JointsMethod and provide KanesMethod what it asks for rather than having a hack like this.

Reply:
We can't process `Body` to `RigidBody` or `Particle`, and since `is_particle` and `is_rigidbody` are required in `Body` so I think this is most optimal solution.

Reply:
Done!